lib/datasource.js

```diff
@@ -499,7 +499,15 @@ DataSource.prototype.setup = function(dsName, settings) {
           debug('Connection fails: %s\nIt will be retried for the next request.', err);
         } else {
           g.error('Connection fails: %s\nIt will be retried for the next request.', err);
-          this.emit('error', err);
+          if (settings.preventCrashWhenUnvailable) {
```

Comment:
This setting need to come from connectors. So you might need to add this connector base repo and then a=enable this in redis connector repo.

Reply:
This setting actually comes from the datasource config. So I don't think we need to make any changes to the connectors.
This option can be passed from the datasource config in an app.

Reply:
Let's rename it to catchFailure

Reply:
Hi, @samarpanB. I just made the changes. Can we merge the PR now?